Resources/public/js/views/services/ez-contenteditviewservice.js

```diff
@@ -40,24 +44,23 @@ YUI.add('ez-contenteditviewservice', function (Y) {
         _load: function (next) {
             var request = this.get('request'),
                 service = this,
-                languageCode = this.get('languageCode');
+                languageCode = this.get('languageCode'),
+                baseLanguageCode = this.get('baseLanguageCode'),
+                languageCodeForLoadContent = baseLanguageCode ? baseLanguageCode : languageCode;
 
             this.get('version').reset();
-            this._loadContent(request.params.id, languageCode, function () {
+            this._loadContent(request.params.id, languageCodeForLoadContent, function () {
                 var tasks,
-                    version = service.get('version'),
                     content = service.get('content'),
                     resources;
 
-                version.set('fields', content.get('fields'));
-
                 resources = content.get('resources');
 
                 tasks = new Y.Parallel();
 
                 service._loadOwner(resources.Owner, tasks.add());
                 service._loadLocation(resources.MainLocation, tasks.add());
-                service._loadContentType(resources.ContentType, tasks.add());
+                service._setVersionFields(tasks.add());
```

Comment:
the code organization here is far from being clear, this needs to be refactored/reorganized. First, you added a `_setVersionFields` which does a lot more then just setting the fields of the version like checking if the content was loaded in the correct language, loading the content type... Then you also repeat the `languageCodeForLoadContent` in `_load` and in `_setVersionFields`.

So to improve a bit the code, I would keep `_loadContentType` here in parallel with the `_loadOwner` and `_loadLocation` and implement the new logic in the `done` callback a few lines below.

The check on the actual language of the content if we have a baseLanguage should be the first thing done in the `_loadContent` callback (it's an edge case and will happen only if the user tweak the URI manually). And please introduce a method to do that so that we have a self-explanatory name to understand the actual tasks done by the code and a doc block explaining why we need that.

After having done that, `_setVersionFields` should be way smaller and easier to understand and will be able to see if how we can maybe improve its code as well.